chain-impl-mockchain/src/key.rs

```diff
@@ -5,8 +5,9 @@ use cardano::hash;
 use cardano::redeem as crypto;
 use chain_core::property;
 
-/// Public key of the entity.
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+/// Public key of the entity. FIXME: do we need this wrapper around
+/// cardano::redeem::PublicKey?
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
```

Comment:
Same here, no real need for `Ord` traits.

Reply:
There needed to establish an ordering between stake pool IDs (and possible stake keys in the future) for selecting a leader based on stake.

Unlikely private keys, I don't think there is a security issue in having Eq/Ord traits on them, right?

Reply:
It does not feel like it's an inherent feature of the public key to be sorted that way. If leadership selection needs this weird way to sort keys, it can use a custom sorter.

I may be overthinking things here, though.

Reply:
Well, what would be a non-weird way to sort them? Lexicographical sort seems like the most straightforward thing to do.

Reply:
I mean that it does not make an inherent sense to sort keys in general, that's why the `PartialOrd/Ord` trait impls look dodgy.

Reply:
That's true, but OTOH, the ability to use public keys or cryptographic hashes as keys in HashMaps/BTreeMaps is pretty useful and arises quite a lot (since we often use them as identities). Given that Rust (unlike C++) unfortunately lacks the ability to specify custom comparators in containers, putting these types in Eq/Ord is the pragmatic thing to do, IMHO.

Reply:
I agree that it makes total sense to derive `PartialEq, Eq, Hash` for public keys, making them usable in hash tables. And perhaps you are right and `Ord` has little possibility of accidental misuse, but makes the thing qualify for B-trees.